website/src/docs/hotchocolate/defining-a-schema/scalars.md

```diff
@@ -171,278 +215,216 @@ public class Query
+    }
 
-        // define how a native type is parsed into a literal,
-        public override IValueNode ParseValue(object? runtimeValue)
+    // define how a value node is parsed to the native .NET type
+    public override object ParseLiteral(IValueNode valueSyntax,
+        bool withDefaults = true)
+    {
+        if (valueSyntax is StringValueNode stringLiteral &&
+            _validator.ValidateCreditCard(stringLiteral.Value))
         {
-            if (runtimeValue is string s &&
-                _validator.ValidateCreditCard(s))
-            {
-                return new StringValueNode(null, s, false);
-            }
-
-            throw new SerializationException(
-                "The specified value has to be a credit card number in the format " +
-                "XXXX XXXX XXXX XXXX");
+            return stringLiteral.Value;
         }
 
-        public override IValueNode ParseResult(object? resultValue)
-        {
-            if (resultValue is string s &&
-                _validator.ValidateCreditCard(s))
-            {
-                return new StringValueNode(null, s, false);
-            }
-
-            throw new SerializationException(
-                "The specified value has to be a credit card number in the format " +
-                "XXXX XXXX XXXX XXXX");
-        }
+        throw new SerializationException(
+            "The specified value has to be a credit card number in the format "
+                + "XXXX XXXX XXXX XXXX",
+            this);
+    }
 
-        public override bool TrySerialize(object? runtimeValue, out object? resultValue)
+    // define how the .NET type is parsed to a value node
+    public override IValueNode ParseValue(object? runtimeValue)
+    {
+        if (runtimeValue is string s &&
+            _validator.ValidateCreditCard(s))
         {
-            if (runtimeValue is string s &&
-                _validator.ValidateCreditCard(s))
-            {
-                resultValue = s;
-                return true;
-            }
-
-            resultValue = null;
-            return false;
+            return new StringValueNode(null, s, false);
         }
 
-        public override bool TryDeserialize(object? serialized, out object? value)
-        {
-            if (serialized is string s &&
-                _validator.ValidateCreditCard(s))
-            {
-                value = s;
-                return true;
-            }
-
-            value = null;
-            return false;
-        }
+        throw new SerializationException(
+            "The specified value has to be a credit card number in the format "
+                + "XXXX XXXX XXXX XXXX",
+            this);
     }
-```
-
-# Additional Scalars
-
-HotChocolate provides additional scalars for more specific usecases.
 
-To use these scalars you have to add the package `HotChocolate.Types.Scalars`
-
-```csharp
-dotnet add package HotChocolate.Types.Scalars
-```
+    // todo: when is this useful? + add comment
```

Comment:
The scalars are at the moment to complex ... I think we can simplify them. This particular method allows the execution engine to more efficiently coerce variables by circumventing the runtime value creation. I will have a look at this all I think next week.